editor/src/templates/editor.tsx

```diff
@@ -402,12 +402,16 @@ export class Editor {
     }
   }
 
-  resetStateOnBlur = () => {
+  resetStateOnBlur = (): void => {
+    const shouldCommitInteraction =
+      this.canvasStore.getState().editor.canvas.interactionSession?.interactionData.type ===
+      'STATIC_REPARENT'
+
     this.boundDispatch(
       [
         EditorActions.clearHighlightedViews(),
         CanvasActions.clearDragState(false),
-        CanvasActions.clearInteractionSession(false),
+        CanvasActions.clearInteractionSession(shouldCommitInteraction),
```

Comment:
TBH I think this could just be always true. I wouldn't want my keyboard interaction to be reset once I'm done keyboarding and blur the editor!